src/main/java/common/tileentities/GTMTE_LapotronicSuperCapacitor.java

```diff
@@ -314,14 +340,16 @@ public boolean checkMachine(IGregTechTileEntity thisController, ItemStack guiSlo
 
 		// Calculate total capacity
 		capacity = BigInteger.ZERO;
-		for(int i = 0; i < capacitors.length; i++){
-			if(i <= 3){
-				final long c = (long) (100000000L * Math.pow(10, i));
-				capacity = capacity.add(
-						BigInteger.valueOf(c).multiply(BigInteger.valueOf(capacitors[i])));
-			} else {
-				capacity = capacity.add(
-						MAX_LONG.multiply(BigInteger.valueOf(capacitors[i])));
+		for(int i = 0; i < capacitors.length; i++){	
+			switch(i) {
+			case 0: capacity = capacity.add(BigInteger.valueOf(100000000L).multiply(BigInteger.valueOf(capacitors[i]))); break;
+			case 1: capacity = capacity.add(BigInteger.valueOf(1000000000L).multiply(BigInteger.valueOf(capacitors[i]))); break;
+			case 2: capacity = capacity.add(BigInteger.valueOf(10000000000L).multiply(BigInteger.valueOf(capacitors[i]))); break;
+			case 3: capacity = capacity.add(BigInteger.valueOf(100000000000L).multiply(BigInteger.valueOf(capacitors[i]))); break;
+			case 4: capacity = capacity.add(MAX_LONG.multiply(BigInteger.valueOf(capacitors[i]))); break;
+			case 5: capacity = capacity.add(BigInteger.valueOf(0L).multiply(BigInteger.valueOf(capacitors[i]))); break;
```

Comment:
This line is no-op.

Reply:
By no-op I guess you mean there doesn't need to be an operation?

Reply:
``BigInteger.valueOf(0L).multiply(BigInteger.valueOf(capacitors[i]))``

=

``BigInteger.valueOf(0L) * capacitors.value()``

=

``BigInteger.valueOf(0L)``

and since this adds this value to the total capacity, you can just remove it. (default will catch it)

Reply:
I realized that after looking at it, but I was unfamiliar with the terminology. Or do you mean removing the case entirely?